### ISSUE TITLE
Provide history of previous searches

### ISSUE THREAD

**Issue by romix** [zealdocs/zeal#597]
Zeal provides back and forward buttons to navigate through the history already.

But it would be nice to have an explicit search history which would list the past search queries. This would allow to jump to one of the old searches very quickly without retyping the search query. 

The implementation of this feature should be pretty trivial in terms of required effort, I guess.

**Comment by trollixx** [comment]
You are correct in regards to the technical implementation, but what about UX? I can't quickly think of an intuitive way to navigate through recent searches.

**Comment by romix** [comment]
In terms of UX I was thinking about something like a drop-down list with previous searches + incremental completion suggestions based on the history as you type your search string.
